src/lib/BridgeBase.sol

```diff
@@ -187,6 +179,8 @@ abstract contract BridgeBase is Receiver, OwnableUpgradeable, UUPSUpgradeable {
         if (state_with_proof.state.epoch != appliedEpoch + 1) {
             revert NotSuccessiveEpochs({epoch: appliedEpoch, nextEpoch: state_with_proof.state.epoch});
         }
+        // increment applied epoch before applying the state to avoid reentrancy
+        ++appliedEpoch;
```

Comment:
So here if we do this before any action, the function is by itself reentrancy safe. However, it doesn't hurt.

Reply:
Removed reentrancyGuard. Please check where should we also modify the code to make it reentrancy safe or where guard should be added